sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs

```diff
@@ -0,0 +1,161 @@
+        ..Default::default()
+    };
+
+    transport.apply_pin_strapping("ROM_BOOTSTRAP")?;
+    let reset_delay = opts
+        .init
+        .bootstrap
+        .options
+        .reset_delay
+        .unwrap_or(Duration::from_millis(50));
+    transport.reset_target(reset_delay, true)?;
+
+    // Now watch the console for the exit conditions.
+    let mut stdout = std::io::stdout();
+    let result = console.interact(&*uart, None, Some(&mut stdout))?;
+    match result {
+        ExitStatus::ExitSuccess => {}
+        _ => {
+            bail!("FAIL: {:?}", result);
+        }
+    };
+    Ok(())
+}
+
+fn test_jedec_id(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let spi = transport.spi("0")?;
+    let id = SpiFlash::read_jedec_id(&*spi, 16)?;
+    log::info!("Read jedec id: {:x?}", id);
+    let mut index = 0;
+    while id[index] == 0x7f {
+        index += 1;
+    }
+    let page = index;
+    let manufacturer = id[index];
+    let product = id[index + 1];
+    let density = id[index + 2];
+
+    assert_eq!(page, opts.jedec_page);
+    assert_eq!(manufacturer, opts.jedec_id);
+    assert_eq!(product, opts.jedec_product);
+    assert_eq!(1u32 << density, opts.flash_size);
+    Ok(())
+}
+
+fn test_sfdp(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let spi = transport.spi("0")?;
+    let sfdp = SpiFlash::read_sfdp(&*spi)?;
+    log::info!("SFDP: {:#x?}", sfdp);
+
+    // We expect a SFDP header of version 1.10 with no additional phdrs
+    // except the mandatory JDEC phdr.
+    assert_eq!(sfdp.header.signature, 0x50444653);
+    assert_eq!(sfdp.header.minor, 0xa);
+    assert_eq!(sfdp.header.major, 0x1);
+    assert_eq!(sfdp.header.nph, 0x0);
+    assert_eq!(sfdp.header.reserved, 0xff);
+
+    // Check the parameters of the mandatory phdr.
+    assert_eq!(sfdp.phdr[0].id, 0x0);
+    assert_eq!(sfdp.phdr[0].minor, 0x7);
+    assert_eq!(sfdp.phdr[0].major, 0x1);
+    assert_eq!(sfdp.phdr[0].dwords, 0x17);
+    assert_eq!(sfdp.phdr[0].offset, 0x10);
+
+    // Check the values of the JEDEC parameters.
+    assert_eq!(sfdp.jedec.block_erase_size, BlockEraseSize::Block4KiB);
+    assert_eq!(
+        sfdp.jedec.write_granularity,
+        WriteGranularity::Granularity64Bytes
+    );
+    assert_eq!(sfdp.jedec.write_en_required, true);
+    assert_eq!(sfdp.jedec.write_en_opcode, 0x50);
+    assert_eq!(sfdp.jedec.erase_opcode_4kib, 0x20);
+    assert_eq!(sfdp.jedec.support_fast_read_112, false);
+    assert_eq!(sfdp.jedec.address_modes, SupportedAddressModes::Mode3b);
+    assert_eq!(sfdp.jedec.support_double_rate_clocking, false);
+    assert_eq!(sfdp.jedec.support_fast_read_122, false);
+    assert_eq!(sfdp.jedec.support_fast_read_144, false);
+    assert_eq!(sfdp.jedec.support_fast_read_114, false);
+    assert_eq!(sfdp.jedec.support_fast_read_222, false);
+    assert_eq!(sfdp.jedec.support_fast_read_444, false);
+    assert_eq!(sfdp.jedec.density, opts.flash_size);
```

Comment:
shouldn't this part check the entire table as defined [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/drivers/spi_device.c;drc=d425d553f22422679d93f87551c3e49b6b920c1a;l=146) (including `0` and `1` bits for unsupported features)?

If the sfdp table does not have those fields yet, we can add a todo to revisit this.

Reply:
AFAICT, this is the entire set of supported bits from the 2011 version of the table.

This check is post-parse, so any undefined fields (from the 2011 spec) are ignored.
I'm also ignoring the parameters for the various modes (112, 122, 144, etc) because they're declared as unsupported by the `support` bits.

I can update the SFDP parser to a newer version of the spec in a separate PR and then come back and update these checks appropriately.